packages/@aws-cdk/aws-ecs-patterns/lib/ecs/application-load-balanced-ecs-service.ts

```diff
@@ -70,27 +80,42 @@ export class ApplicationLoadBalancedEc2Service extends ApplicationLoadBalancedSe
   /**
    * Constructs a new instance of the ApplicationLoadBalancedEc2Service class.
    */
-  constructor(scope: Construct, id: string, props: ApplicationLoadBalancedEc2ServiceProps) {
+  constructor(scope: Construct, id: string, props: ApplicationLoadBalancedEc2ServiceProps = {}) {
     super(scope, id, props);
 
-    this.taskDefinition = new Ec2TaskDefinition(this, 'TaskDef', {
-      executionRole: props.executionRole,
-      taskRole: props.taskRole
-    });
+    if (props.taskDefinition && props.taskImageOptions) {
+      throw new Error('You must specify either a taskDefinition or taskImageOptions, not both.');
+    } else if (props.taskDefinition) {
+      this.taskDefinition = props.taskDefinition;
+    } else if (props.taskImageOptions) {
+      const taskImageOptions = props.taskImageOptions;
+      this.taskDefinition = new Ec2TaskDefinition(this, 'TaskDef', {
+        executionRole: taskImageOptions.executionRole,
+        taskRole: taskImageOptions.taskRole
+      });
 
-    const containerName = props.containerName !== undefined ? props.containerName : 'web';
-    const container = this.taskDefinition.addContainer(containerName, {
-      image: props.image,
-      cpu: props.cpu,
-      memoryLimitMiB: props.memoryLimitMiB,
-      memoryReservationMiB: props.memoryReservationMiB,
-      environment: props.environment,
-      secrets: props.secrets,
-      logging: this.logDriver,
-    });
-    container.addPortMappings({
-      containerPort: props.containerPort || 80
-    });
+      // Create log driver if logging is enabled
+      const enableLogging = taskImageOptions.enableLogging !== undefined ? taskImageOptions.enableLogging : true;
+      const logDriver = taskImageOptions.logDriver !== undefined
+                          ? taskImageOptions.logDriver : enableLogging
+                            ? this.createAWSLogDriver(this.node.id) : undefined;
```

Comment:
Apologies for the naive question, but how does this double ternary work? I personally find it hard to read but not sure if this is idiomatic in the rest of the code.

Reply:
I believe I've seen it elsewhere in the CDK. It's very common in typescript/javascript though.